### ISSUE TITLE
Duplicated item when created with a template.

### ISSUE THREAD

**Issue by AGagarin** [kidd/org-gcal.el#68]
Hi i have a template to create Google Appointment in emacs.
`*** Google Appointment
#+BEGIN_SRC emacs-lisp
(after! org (add-to-list 'org-capture-templates
             '("a" "Appointment" entry (file  "~/.org/gcal.org" )
"* %?
:PROPERTIES:
:ETag:
:calendar-id: anpbox@gmail.com
:ID:
:END:
:org-gcal:\n%^T
:END:")))
#+END_SRC
`
But when i create an appointment it gets duplicated.
The appointment created with template
`* Test for GitHub
:PROPERTIES:
:ETag:
:calendar-id: anpbox@gmail.com
:ID:
:END:
:org-gcal:
<2020-04-13 Mon 13:26>
:END:
`
Duplicated one created just after C-c C-c
`* Test for GitHub
:PROPERTIES:
:ETag:     "3173201679274000"
:calendar-id: anpbox@gmail.com
:ID:       chgr74g0kuhlb80urpk3qutmhc/anpbox@gmail.com
:END:
:org-gcal:
<2020-04-13 Mon 13:26>
:END:
`

**Comment by andreasqueri** [comment]
Same issue here. Create an event from template create 1 event on google calendar but 2 event in org file

**Comment by telotortium** [comment]
What exactly are you running? Are you calling `org-gcal-sync`, `org-gcal-fetch`, or `org-gcal-post-at-point`?

**Comment by andreasqueri** [comment]
I have this in my config:
`(use-package org-gcal
:ensure t
:config
(setq org-gcal-client-id "3434234234324210-dgjd3981e9rttm4j1dkvmqdvbl77dn96.apps.googleusercontent.com"
org-gcal-client-secret "fgdfgdfgdfgdfgdfgsE9"
org-gcal-file-alist '(("andrea.squeri@gmail.com" . "/home/andre/org/Beast/gcal.org"))))
(add-hook 'org-agenda-mode-hook (lambda () (org-gcal-sync) ))
(add-hook 'org-capture-after-finalize-hook (lambda () (org-gcal-sync)))`

and i have this capture template:
`("g" "Google Calendar" entry (file  "~/org/Beast/gcal.org" )
                                "* %?\n:PROPERTIES:\n:calendar-id: andrea.squeri@gmail.com\n:END:\n:org-gcal:\n%^T\n:END:")`

When i finish the capture with C-c C-c i got an error "deferred error : (wrong-type-argument stringp nil)" so the capture template is written in the gcal.org without the data fetch from google calendar(id,etags), but the event is created in the google calender. Then when is executed the sync (for example when i start the org agenda) the event is refetched from google calendar in the gcal.org , so i have 1 google event and two "org event".

`* dsdssdd
:PROPERTIES:
:calendar-id: andrea.squeri@gmail.com
:END:
:org-gcal:
<2020-06-20 sab 10:00-11:00>
:END:

* dsdssdd
:PROPERTIES:
:ETag:     "3185226469498000"
:calendar-id: andrea.squeri@gmail.com
:ID:       eaftqkn6tl5d227m3qef2343bk/andrea.squeri@gmail.com
:END:
:org-gcal:
<2020-06-20 sab 10:00-11:00>
:END:`

The first entry is written by capture template, it is written also in the google calendar but because the error "deferred error : (wrong-type-argument stringp nil)" the data from the calendar aren't fetched.
When i syncronize with org-gcal-sync, the second entry is written so i have 2 item.
If i put the same entry directly in the file (not from capture) and the execute org-gcal-post-at-point works fine

**Comment by telotortium** [comment]
The behavior of org-gcal-sync here is a little unexpected to me - I'll have
to look into that.

For now, I'd advise you to change your org-capture to run
org-gcal-post-at-point after capturing the calendar event. I think
org-capture templates can do this, or else tell your finalize hook to post
the event if it has a calendar id property.

**Comment by andreasqueri** [comment]
If i disable the hook in org-gcal.el file 
`(defun org-gcal--capture-post ()
  
  (dolist (i org-gcal-fetch-file-alist)
    (when (string=  (file-name-nondirectory (cdr i))
		    (substring (buffer-name) 8))
      (org-gcal-post-at-point))))

;;(add-hook 'org-capture-before-finalize-hook 'org-gcal--capture-post)`

and before finalize the capture with C-c C-c i manually call the org-gcal-post-at-point in the template buffer it work. I don't know how call org-gcal-post-at-point automatically before finalize the template

**Comment by vovav80** [comment]
I have exactly the same issue. An event is created in Google calendar and then syncing creates an additional event in org with with an etag and id, whilst keeping the original event.

**Comment by andreasqueri** [comment]
I Belice the problem Is the error"wrong-type-argument stringp nil) " when finish the capture. Because of this the entro Is written in the Google calendar and in the org file, but in the org file aren't fetched the google data( id) so when you Sync with Google, in the org file Is added the entro from the Google calendar with the id. And the result Is that in the org file are presente 2 item, one in Sync with Google and One not in sync

**Comment by ugurbolat** [comment]
yup same issue with org-gcal-sync when using org-capture templates

**Comment by telotortium** [comment]
@ugurbolat Can you try updating to `HEAD` on `master` and see if #106 has fixed the issue for you?

**Comment by ugurbolat** [comment]
@telotortium yup, updated yesterday and it solved the bug. thank you!

**Comment by jlaznarte** [comment]
I'm using the last version (emacs.d/elpa/develop/org-gcal-20210422.3) and it is still happening to me. :( I'm a total newbie, so I don't really know how to report it. But I'm willing to help.
